File: agent-code-platform/app/model/event/ai_task_event.py
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class TaskInfo(BaseModel):
    task_id: str = Field(alias="taskId")
    user_id: str = Field(alias="userId")
    project_type: str = Field(alias="projectType")
    preview_path: str = Field(alias="previewPath")
    app_id: str | None = Field(default=None, alias="appId")
    task_type: str = Field(default="GENERATE", alias="taskType")
    base_version_id: str | None = Field(default=None, alias="baseVersionId")
    target_version_id: str | None = Field(default=None, alias="targetVersionId")
    base_source_path: str | None = Field(default=None, alias="baseSourcePath")
    target_source_path: str | None = Field(default=None, alias="targetSourcePath")

    model_config = ConfigDict(populate_by_name=True)


class Payload(BaseModel):
    prompt: str
    context_messages: list[Any] = Field(default_factory=list, alias="contextMessages")
    selected_elements: list[Any] = Field(default_factory=list, alias="selectedElements")
    scope: str = "single"

    model_config = ConfigDict(populate_by_name=True)


class AiTaskEvent(BaseModel):
    event_id: str = Field(alias="eventId")
    timestamp: int
    trace_id: str = Field(alias="traceId")
    task: TaskInfo
    payload: Payload

    model_config = ConfigDict(populate_by_name=True)

File: src/main/java/com/chg/yuaicodemother/kafka/AiTaskEvent.java
package com.chg.yuaicodemother.kafka;

import lombok.Builder;
import lombok.Data;
import java.util.List;

@Data
@Builder
public class AiTaskEvent {
    private String eventId;     // 消息全局唯一ID，用于防重
    private Long timestamp;     // 毫秒级时间戳
    private String traceId;     // 链路追踪 ID
    private TaskInfo task;      // 任务核心信息
    private Payload payload;    // 具体载荷

    @Data
    @Builder
    public static class TaskInfo {
        private String taskId;
        private String userId;
        private String appId;
        private String previewPath;
        private String projectType; // 智能路由结果: HTML/VUE_PROJECT 等
        private String taskType; // GENERATE / EDIT
        private String baseVersionId;
        private String targetVersionId;
        private String baseSourcePath;
        private String targetSourcePath;
    }

    @Data
    @Builder
    public static class Payload {
        private String prompt;
        private List<Object> contextMessages;
        private List<Object> selectedElements;
        private String scope;
    }
}
